platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventCreatorThread.java

```diff
@@ -65,13 +65,13 @@ public EventCreatorThread(
         this.random = random;
         this.otherNodes = StreamSupport.stream(addressBook.spliterator(), false)
                 // don't create events with self as other parent
-                .filter(a -> !selfId.equalsMain(a.getId()))
-                .map(a -> NodeId.createMain(a.getId()))
+                .filter(a -> !selfId.matches(a.getId()))
```

Comment:
is overloading equals to compare a NodeId to a primitive long appropriate?  I will get a null pointer exception otherwise because the long gets converted into a boxed Long and the class comparison will fail.   But overloading equals will work, I just don't know if it is confusing.

Reply:
Ah!  I see what you mean here.

Reply:
turns out `a` is an address and is not comparable with `equals`.